### ISSUE TITLE
Is IR being maintained any longer?

### ISSUE THREAD

**Issue by avalanche1** [iron-meteor/iron-router#1348]
latest release half a year ago, 240 open issues... some announcement from project authors maybe?

**Comment by gabrielhpugliese** [comment]
* 865 commits on devel branch
* 22 branches
* 53 contributors (including big players from community)
* 150k downloads
* Last commit on devel was 18 days ago

I don't think something that big can die so easily.

**Comment by copleykj** [comment]
Yeah, I think this is really just a time issue. It's something that most FOSS projects suffer from, especially the smaller ones.. It takes money to free developers time from their other paid commitments.

**Comment by VladShcherbin** [comment]
We will all migrate to Flow Router, the same thing happened to Meteor Router before IR.

**Comment by gabrielhpugliese** [comment]
That would be very sad, again. Meteor Router was very little when that happened.

**Comment by dandv** [comment]
#1219 suggests the IR team no longer has time to actively review PRs :(

**Comment by allanlundhansen** [comment]
@cmather I understand and appreciate the challenges regarding time management of ressources between open source and a company, that being said I do believe (even with regards to your business that from my understanding relates quite a bit to the development community) it would be prudent with an update... Just any kind... To the community that also could be potential customers and referrals to your business.

**Comment by cmather** [comment]
I'll post an update and plans soon.

**Comment by cmather** [comment]
So the short of it is I'm just under water with teaching and running the EM business, which it turns out is a full time job. I'm very sorry for keeping folks hanging. I feel really bad about it. 

I'm putting out a new release in a few minutes that contains a lot of the pull requests. I've been trying to work with MDG Core for a while now to work a router into core (build or integrate) which I think is the right thing to do. I'll write more about that a little later. In the mean time I don't see why starting a router from scratch makes a lot of sense. You can't get much simpler than this:

```javascript
Router.route('/path/:id', function () {
  this.render('MyTemplate', {data: {someContext: true}});
});
```

Want things not to be reactive:

```javascript
Router.route('/path/:id', function () {
  var self = this;
  Tracker.nonreactive(function () {
    self.render('SomeTemplate');
  });
});
```

The routing package is quite simple. But it's supported by several other packages that were required  in order to make it work ( dynamic template, dynamic layout, url, location, etc). You can replace any of those layers as desired. It's just a matter of building it.

**Comment by arunoda** [comment]
Hi @cmather it's nice to see IR comes back to alive. I can understand the weight you have. 

But flow is not to replace IR. We've our own set of rules and how things will be done.
We build our APIs in that way. We help users to write better route code by restricting them into a set of rules and enforcing them via our APIs.

Let me tell me few of these:

#### No rendering/layout support
Meteor's dynamic templates are great. We enforce users to use it by that restriction

#### No waitOn
We don't have waiton, so users put that logic into template layer and makes their apps more responsive

#### No data context (on the router)
Getting data context on the router leads to a lot of re-renders. There are many ways to get around to this problem.

#### Reactive Apis
We have a sweet reactive API which allows you to get and set stuff to the router. Based on that I've suggested written a blog post: https://meteorhacks.com/meteor-ui-pattern-keeping-app-state-in-the-url

**Problmes**

The only problem right now is how to handle auth related logic like redirects and so on. We don't allow to use reactive content of middlewares. So, this is kind a tricky right now. So, I'm working on a new project called `auth-manager` to decouple that from the router and put it into the template layer.

--- 
So basically, what I wanted to say IR is pretty good and ways to extend that. People could use it in the right way or wrong way(still subjective). But we enforce users to do in the right way.

>  In the mean time I don't see why starting a router from scratch makes a lot of sense

I don't agree with this. We couldn't experiment with above stuff if we've haven't started with from the root. You should not think IR will be the only routing solution we should have. Having more solution helps developers. Specially, since we are not a copy cat of IR.

**Comment by avalanche1** [comment]
@cmather, Why don't you release IR maintenance into full community support if you don't have time for it? I mean authorize a group of volunteers to merge PRs, respond to issues and so forth?

**Comment by SachaG** [comment]
@avalanche1 what's preventing people from doing all that right now? I don't think @cmather is preventing anybody from responding to issues? And if you're willing to take time to learn IR inside out so you can properly test and merge PRs, I don't think Chris would object to that either :)

**Comment by arunoda** [comment]
I agree with @SachaG. Maintaining a such an important project is not simple. Someone might send a PR for a one fix. That may lead to ton of other issues. So, it's not simply as taking this over to someone else.

There should be one person overlooking the whole process. And that's hard and time consuming.
Automated tests and can helps a lot, still not 100% sure until you get request from real use cases.

**Comment by gabrielhpugliese** [comment]
I think what prevents it is that he is the only owner of the organization

![image](https://cloud.githubusercontent.com/assets/1252818/7950967/f7878174-0977-11e5-8046-4048defb64d5.png)

**Comment by arunoda** [comment]
I don't think so. There are many, including me.
What is really happening is in GH organzations, users need to expose them to the public. Otherwise they are hidden :)

**Comment by gabrielhpugliese** [comment]
Oh I see :) Sorry!

**Comment by avalanche1** [comment]
Why don't expand project owners list with trusted ppl who are willing and have the time and the ability to devote to prj development? I saw some PR not being merged for 3+ months. That's hardly  motivating to submit further PRs when ur previous ones are not being merged.
